### PR TITLE
feat: Add example of how to integrate Camoufox into PlaywrightCrawler

### PR DESCRIPTION
**Pull request by Pijukatel**
### Description

Show how to integrate `Camoufox` into `PlaywrightCrawler`. 
Added one docs page and one example code.
Added `playwright-camoufox` template to cli call `crawlee create`.
Fixed wrong template indentation when `Apify integration` was selected in cli.

### Issues

- Closes: https://github.com/apify/crawlee-python/issues/684

### CODE REVIEW

**Comment by Pijukatel** [comment]
Camoufox template works locally, but fails when used in Actor on platform. That is Camoufox problem, but still it would be good to solve if we have it as template:
Issue:
https://github.com/daijro/camoufox/issues/125
Proposed fix:
https://github.com/daijro/camoufox/pull/126/files

File: src/crawlee/project_template/templates/main.py [hunk 1]
```diff
@@ -20,7 +20,11 @@
 # % endif
 # % endblock
 # % endfilter
+{% if self.pre_main is defined %}
```
Comment (janbuchar) [inline src/crawlee/project_template/templates/main.py:23]: Interesting trick! Could you use line expressions (`# % if ...`) here like in the rest of the file?

File: docs/examples/code/playwright_crawler_with_camoufox.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    # Define a hook that will be called each time before navigating to a new URL.
+    # The hook receives a context parameter, providing access to the request and
+    # browser page among other things. In this example, we log the URL being
+    # navigated to.
+    @crawler.pre_navigation_hook
+    async def log_navigation_url(context: PlaywrightPreNavigationContext) -> None:
+        context.log.info(f'Navigating to {context.request.url} ...')
+
+    # Run the crawler with the initial list of URLs.
+    await crawler.run(['https://news.ycombinator.com/'])
```
Comment (janbuchar) [inline docs/examples/code/playwright_crawler_with_camoufox.py:78]: There is a lot of stuff going on that could distract the reader from the actual gist of the example - could we pare it down a bit?

File: docs/examples/playwright_crawler_with_camoufox.mdx
Comment (janbuchar) [inline docs/examples/playwright_crawler_with_camoufox.mdx:1]: Can you mention the new template here as well?

Comment (Pijukatel) [reply docs/examples/playwright_crawler_with_camoufox.mdx:1]: Done

File: docs/examples/code/playwright_crawler_with_camoufox.py (hunk 2 above)
Comment (Pijukatel) [reply docs/examples/code/playwright_crawler_with_camoufox.py]: I reduced most of the example and kept only minimal example that still demonstrates the use of the crawler.

File: docs/examples/code/playwright_crawler_with_camoufox.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import asyncio
+
+#  This external package needs to be installed. It is not included in crawlee.
+from camoufox import AsyncNewBrowser  # type: ignore[import-not-found]
```
[the remark this answers is not in the document]
Comment (Pijukatel) [reply docs/examples/code/playwright_crawler_with_camoufox.py]: Done

File: src/crawlee/project_template/templates/main_camoufox.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+from camoufox import AsyncNewBrowser
+from typing_extensions import override
+
+from crawlee._utils.context import ensure_context
+from crawlee.browsers import PlaywrightBrowserPlugin, PlaywrightBrowserController, BrowserPool
+from crawlee.playwright_crawler import PlaywrightCrawler
+# % endblock
+
+# % block pre_main
+class ExampleCamoufoxPlugin(PlaywrightBrowserPlugin):
```
[the remark this answers is not in the document]
Comment (Pijukatel) [reply src/crawlee/project_template/templates/main_camoufox.py]: Done

File: src/crawlee/project_template/templates/main.py (hunk 1 above)
Comment (Pijukatel) [reply src/crawlee/project_template/templates/main.py]: Done

File: src/crawlee/project_template/cookiecutter.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
     "project_name": "crawlee-python-project",
     "__package_name": "{{ cookiecutter.project_name|lower|replace('-', '_') }}",
-    "crawler_type": ["beautifulsoup", "parsel", "playwright"],
+    "crawler_type": ["beautifulsoup", "parsel", "playwright", "camoufox"],
```
[the remark this answers is not in the document]
Comment (Pijukatel) [reply src/crawlee/project_template/cookiecutter.json]: Done

**Review by janbuchar — APPROVED** [review]

**Pull request merged**